org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/extensions/XMLExtensionsRegistry.java

```diff
@@ -64,8 +65,11 @@
 	private IXMLNotificationService notificationService;
 
 	private final Map<Class, Object> components;
+	
+	private final XMLLanguageServer server;
 
-	public XMLExtensionsRegistry() {
+	public XMLExtensionsRegistry(XMLLanguageServer server) {

```

Comment:
I would like to avoid having dependency to the XMLLanguageServer.

Reply:
Yes, i thought so :-) I'll try to come with some "proxy" API to avoid exposing LS directly